### PR TITLE
Add lokalise-file-pull action

### PR DESCRIPTION
**Pull request by igorDolzh**
I added `lokalise-file-pull` action

With the help of this action, it will be possible to fetch translations from the localise.com from PR directly to the project

# localise-file-pull

Lets you automatically pull messages from your localise.co project to your repository

## How to use

```yaml
name: localise-file-pull

on:
  push:
    # Only run workflow for pushes to specific branches
    branches:
      - master
    # Only run workflow when matching files are changed
    paths:
      - "src/locales/*/messages.po"

jobs:
  upload:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - uses: igorDolzh/localise-file-pull@v10.0
        with:
          # Api token for the Lokalise account
          # with read/write access to the project
          api-token: ${{ secrets.LOCALIZE_TOKEN }}

          # ID of the project to sync
          project-id: project-id

          # The relative file path where language files will be found
          file-path: src/locales/%LANG_ISO%/messages.po

          # Download options for https://app.lokalise.com/api2docs/curl/#transition-download-files-post
          download-options: '{"export_empty_as": 'skip'}'
```

### CODE REVIEW

File: lokalise-file-pull/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import { LokaliseApi, DownloadFileParams, Keyable } from '@lokalise/node-api'
```
Comment (igorDolzh) [inline lokalise-file-pull/src/main.ts:1]: This is the actual file to check

**Review by bel0v — APPROVED** [review]

File: lokalise-file-pull/action.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Lokalise Pull Messages
+description: "Pushes files to Lokalise"
```
Comment (Wiingaard) [inline lokalise-file-pull/action.yml:2]: `Pulls files from Lokalise`

File: lokalise-file-pull/action.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+branding:
+  icon: arrow-up-circle
```
Comment (Wiingaard) [inline lokalise-file-pull/action.yml:4]: Should probably be `arrow-down-circle`?

File: lokalise-file-pull/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+async function run() {
+    const lokalise: LokaliseApi = new LokaliseApi({
+        apiKey
+    })
+
+    const getZipFile = (fileUrl: string): Promise<AdmZip> => {
+        return new Promise((resolve, reject) => {
+            const req = https.get(fileUrl, async (res) => {
+                const data = [] as any
+                let dataLen = 0
```
Comment (Wiingaard) [inline lokalise-file-pull/src/main.ts:23]: How about:
```suggestion
                let totalDataLength = 0
```

File: lokalise-file-pull/src/main.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            fs.mkdirSync(dirname, { recursive: true });
+        }
+        fs.writeFileSync(file, data);
+    };
+
+    let options = {}
+    try {
+        options = downloadOptions ? JSON.parse(downloadOptions) : {}
+    } catch {
+        options = {}
```
Comment (Wiingaard) [inline lokalise-file-pull/src/main.ts:62]: Maybe you wanna add a log here that's it's ignoring `downloadOptions` because they couldn't be parsed?

**Review by Wiingaard — APPROVED** [review]
LGTM 👍

**Pull request merged**